source/templates.tex

```diff
@@ -2836,36 +2832,12 @@
 the \grammarterm{init-capture} pack.
 \end{itemize}
 
-All of the $\mathtt{E}_i$ become items in the enclosing list.
-\begin{note}
-The variety of list varies with the context:
-\grammarterm{expression-list},
-\grammarterm{base-specifier-list},
-\grammarterm{template-argument-list}, etc.
-\end{note}
-When $N$ is zero, the instantiation of the expansion produces an empty list.
-Such an instantiation does not alter the syntactic interpretation of the
-enclosing construct, even in cases where omitting the list entirely would
-otherwise be ill-formed or would result in an ambiguity in the grammar.
-\begin{example}
-\begin{codeblock}
-template<class... T> struct X : T... { };
-template<class... T> void f(T... values) {
-  X<T...> x(values...);
-}
-
-template void f<>();    // OK: \tcode{X<>} has no base classes
-                        // \tcode{x} is a variable of type \tcode{X<>} that is value-initialized
-\end{codeblock}
-\end{example}
-
 \pnum
 The instantiation of a \tcode{sizeof...} expression\iref{expr.sizeof} produces
-an integral constant containing the number of elements in the pack
-it expands.
+an integral constant with value $N$.
```

Comment:
Should we say that `sizeof...(Pack)` expands the elements of the pack? I think that this isn't observable, even in the weird alias template cases where you end up with `sizeof...` applied to something that's not just the name of a pack, so this is probably fine, but it seems slightly surprising. Maybe the wording consistency is worth the modicum of surprise.

Reply:
p7 status quo: "A pack whose name appears within the pattern of a pack expansion is expanded by that pack expansion."

What else do you want me to say?

Reply:
Hmm, I had interpreted that "is expanded by" as being a simple boolean property of the appearance of the pack, tying into the rule that all appearances of a pack need to be expanded, and not anything to do with substitution of the elements of the pack into some enclosing pattern.

But I think it's fine to leave this as you have it. Sure, we'll substitute into the *identifier* and produce a bunch of *id-expression*s / *template-name*s/*typedef-name*s naming the elements of the pack, then count them and drop them on the floor, but it doesn't matter that we do so.